adafruit_max31855.py

```diff
@@ -0,0 +1,75 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2017 Radomir Dopieralski for Adafruit Industries.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import ustruct
+
+
+class MAX31855:
+    """
+    Driver for the MAX31855 thermocouple amplifier.
+    """
+
+    def __init__(self, spi, cs):
+        self.spi = spi
+        self.cs = cs
+        self.data = bytearray(4)
+
+    def read(self, internal=False, raw=False):
+        """
+        Read the measured temperature.
+
+        If ``internal`` is ``True``, return a tuple with the measured
+        temperature first and the internal reference temperature second.
+
+        If ``raw`` is ``True``, return the values as 14- and 12- bit integers,
+        otherwise convert them to Celsuius degrees and return as floating point
+        numbers.
+        """
```

Comment:
You don't really have any choice about what you are reading from this chip. There are no registers, you just do a single read and get all the data all at once.

Raw access is useful for debugging or for getting more precision than the puny ~6 decimal digits supported by the float type. It generally opens the driver for use cases that we didn't anticipate, but that can still be useful, giving the user who knows what they are doing more power, but at the same time letting the common use cases stay simple.